src/AppInstallerRepositoryCore/Microsoft/Schema/Checkpoint_1_0/CheckpointDataTable.cpp

```diff
@@ -0,0 +1,190 @@
+
+namespace AppInstaller::Repository::Microsoft::Schema::Checkpoint_V1_0
+{
+    using namespace SQLite;
+    using namespace std::string_view_literals;
+    static constexpr std::string_view s_CheckpointDataTable_Table_Name = "CheckpointData"sv;
+    static constexpr std::string_view s_CheckpointDataTable_CheckpointId_Column = "CheckpointId"sv;
+    static constexpr std::string_view s_CheckpointDataTable_ContextData_Column = "ContextData"sv;
+    static constexpr std::string_view s_CheckpointDataTable_Name_Column = "Name"sv;
+    static constexpr std::string_view s_CheckpointDataTable_Value_Column = "Value"sv;
+    static constexpr std::string_view s_CheckpointDataTable_Index_Column = "Index"sv;
+
+    namespace
+    {
+        SQLite::rowid_t SetNamedValue(SQLite::Connection& connection, std::string_view name, std::string_view value)
+        {
+            SQLite::Builder::StatementBuilder builder;
+            builder.InsertInto(s_CheckpointDataTable_Table_Name)
+                .Columns({ s_CheckpointDataTable_CheckpointId_Column,
+                    s_CheckpointDataTable_ContextData_Column,
+                    s_CheckpointDataTable_Name_Column,
+                    s_CheckpointDataTable_Value_Column,
+                    s_CheckpointDataTable_Index_Column})
+                .Values(name, value);
+
+            builder.Execute(connection);
+            return connection.GetLastInsertRowID();
+        }
+
+        std::string GetNamedValue(SQLite::Connection& connection, std::string_view name)
+        {
+            SQLite::Builder::StatementBuilder builder;
+            builder.Select({ s_CheckpointDataTable_Value_Column })
+                .From(s_CheckpointDataTable_Table_Name).Where(s_CheckpointDataTable_Name_Column).Equals(name);
+
+            SQLite::Statement statement = builder.Prepare(connection);
+            THROW_HR_IF(E_NOT_SET, !statement.Step());
+            return statement.GetColumn<std::string>(0);
+        }
+    }
+
+    std::string_view CheckpointDataTable::TableName()
+    {
+        return s_CheckpointDataTable_Table_Name;
+    }
+
+    void CheckpointDataTable::Create(SQLite::Connection& connection)
+    {
+        using namespace SQLite::Builder;
+
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createCheckpointDataTable_v1_0");
+
+        StatementBuilder createTableBuilder;
+        createTableBuilder.CreateTable(s_CheckpointDataTable_Table_Name).BeginColumns();
+        createTableBuilder.Column(ColumnBuilder(s_CheckpointDataTable_CheckpointId_Column, Type::Int).NotNull());
+        createTableBuilder.Column(ColumnBuilder(s_CheckpointDataTable_ContextData_Column, Type::Int));
+        createTableBuilder.Column(ColumnBuilder(s_CheckpointDataTable_Name_Column, Type::Text));
+        createTableBuilder.Column(ColumnBuilder(s_CheckpointDataTable_Value_Column, Type::Text));
+        createTableBuilder.Column(ColumnBuilder(s_CheckpointDataTable_Index_Column, Type::Int));
```

Comment:
Table should probably have a primary key on { checkpoint, data, name, index }.  Also, I don't think you want to allow nulls in any of the columns except for value, and then I think it would be better to have the design be that you simply don't insert the value rather than putting in a null one.

Reply:
Changed to suggested